### PR TITLE
CharmEditor for polls description

### PR DESCRIPTION
**Pull request by Devorein**
### <samp>🤖 Generated by Copilot at 9cc9b92</samp>

This pull request updates the vote model and components to use `content` and `contentText` fields instead of `description` for storing and displaying vote information. It also fixes some imports, syntax, and testing issues related to the vote schema change. It adds a migration script to transfer the existing vote descriptions to the new fields. It also improves the rendering of the sidebar drawer in the `CharmEditor` component.

### WHY
image
Will have to run the migration, then send another PR to remove the `description` field completely from poll model.

### CODE REVIEW

File: pages/api/votes/index.ts [hunk 1]
```diff
@@ -60,7 +60,26 @@ async function getVotes(req: NextApiRequest, res: NextApiResponse<ExtendedVote[]
...
+      contentText: vote.contentText ?? vote.description,
+      content: vote.content ?? {
+        type: 'doc',
+        content: [
+          {
+            type: 'paragraph',
+            content: [
+              {
+                type: 'text',
+                content: vote.description
```
Comment (mattcasey) [inline pages/api/votes/index.ts:75]: description can be null - we should check it exists, use any empty string or return 'null' for the content

**Comment by mattcasey** [comment]
The placeholder text is hard to see i think we may have addressed this some other component...
image

**Review by mattcasey — CHANGES_REQUESTED** [review]
Chris asked not to include user mentions - makes sense, because users might expect that we send notifications if they are mentioned. This might require adding another prop to charmeditor. I think that's OK - long-term maybe we should be taking an array of 'plugins' to be enabled

Also the video embed doesn't work. I think we should just disable that one too?

File: __integration-tests__/server/pages/api/votes/index.spec.ts [hunk 2]
```diff
@@ -77,7 +70,8 @@ describe('POST /api/votes - Create a new poll', () => {
     });
     const newVote: VoteDTO = {
       deadline: new Date(),
-      description: '',
+      content: {},
```
[the remark this answers is not in the document]
Comment (Devorein) [reply __integration-tests__/server/pages/api/votes/index.spec.ts:73]: Thank you. Updated.

File: pages/api/votes/index.ts (hunk 1 above)
Comment (Devorein) [reply pages/api/votes/index.ts:75]: Good catch. Updated.

File: scripts/migrations/2023_06_23_transferVoteDescription.ts [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+
+// a function that adds OP token as a default payment for all spaces in the database
+async function init() {
+
+  const votes = await prisma.vote.findMany({})
+
+  for (const vote of votes) {
+    await prisma.vote.update({
+      data: {
+        content: {
```
[the remark this answers is not in the document]
Comment (Devorein) [reply scripts/migrations/2023_06_23_transferVoteDescription.ts:11]: Thank you. Updated.

**Pull request merged**